### DISCUSSION TITLE
Memory Limit error when indexing fulltextsearch

### DISCUSSION THREAD

**Discussion by caplam** [nextcloud/all-in-one#4570] [Questions] [answered]
I had previously posted but it seems my post was deleted. So i'll make it shorter.
I saw dozen of topics about memory limits but nothing helpful.
To sumup: indexing with fulltextsearch doesn't work.
I reseted and ran test: all is ok. But when i launch indexing ("sudo -u www-data php occ fulltextsearch:index")
it starts with : `Warning: Failed to set memory limit to 0 bytes (Current memory usage is 2097152 bytes) in Unknown on line 0
The current PHP memory limit is below the recommended value of 512MB.` 
run for 1 or 2 minutes and ends up with:
`PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 31457312 bytes) in /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/elastic/transport/src/Transport.php on line 250`

Following @szaimen advice i setup an env var in nextcloud_aio_mastercontainer: NEXTCLOUD_MEMORY_LIMIT
I set it up to 8192M
I have no idea if it's a limit on a per process basis or for the whole instance or anything else.
It seems to be taken into account:
![Capture d’écran 2024-04-22 à 13 47 29](https://github.com/nextcloud/all-in-one/assets/26901633/c4ad5295-3690-4614-ae97-22f66fc98695)
Containers had been restarted several times.
My instance is running on unraid server.

**Comment by szaimen** [comment]
Hi, see https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-run-occ-commands

**Comment by caplam** [comment]
I think i have not been clear enough.
i ran the command "sudo -u www-data php occ fulltextsearch:index" in nextcloud-aio-nexctloud" container after having resized memory but the error message remains the same:
`it starts with : Warning: Failed to set memory limit to 0 bytes (Current memory usage is 2097152 bytes) in Unknown on line 0 The current PHP memory limit is below the recommended value of 512MB.
run for 1 or 2 minutes and ends up with:
PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 31457312 bytes) in /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/elastic/transport/src/Transport.php on line 250
`
So despite php indicates 8G of memory indexing can not run because it lacks of memory

edit: and here is the result of a test:
`/var/www/html # sudo -u www-data php occ fulltextsearch:test

Warning: Failed to set memory limit to 0 bytes (Current memory usage is 2097152 bytes) in Unknown on line 0
The current PHP memory limit is below the recommended value of 512MB.
 
.Testing your current setup:  
Creating mocked content provider. ok  
Testing mocked provider: get indexable documents. (2 items) ok  
Loading search platform. (Elasticsearch) ok  
Testing search platform. ok  
Locking process ok  
Removing test. ok  
Pausing 3 seconds 1 2 3 ok  
Initializing index mapping. ok  
Indexing generated documents. ok  
Pausing 3 seconds 1 2 3 ok  
Retreiving content from a big index (license). (size: 32386) ok  
Comparing document with source. ok  
Searching basic keywords:  
 - 'test' (result: 1, expected: ["simple"]) ok  
 - 'document is a simple test' (result: 2, expected: ["simple","license"]) ok  
 - '"document is a test"' (result: 0, expected: []) ok  
 - '"document is a simple test"' (result: 1, expected: ["simple"]) ok  
 - 'document is a simple -test' (result: 1, expected: ["license"]) ok  
 - 'document is a simple +test' (result: 1, expected: ["simple"]) ok  
 - '-document is a simple test' (result: 0, expected: []) ok  
 - 'document is a simple +test +testing' (result: 1, expected: ["simple"]) ok  
 - 'document is a simple +test -testing' (result: 0, expected: []) ok  
 - 'document is a +simple -test -testing' (result: 0, expected: []) ok  
 - '+document is a simple -test -testing' (result: 1, expected: ["license"]) ok  
 - 'document is a +simple -license +testing' (result: 1, expected: ["simple"]) ok  
Updating documents access. ok  
Pausing 3 seconds 1 2 3 ok  
Searching with group access rights:  
 - 'license' - [] -  (result: 0, expected: []) ok  
 - 'license' - ["group_1"] -  (result: 1, expected: ["license"]) ok  
 - 'license' - ["group_1","Group_2"] -  (result: 1, expected: ["license"]) ok  
 - 'license' - ["group_3","Group_2"] -  (result: 1, expected: ["license"]) ok  
 - 'license' - ["group_3"] -  (result: 0, expected: []) ok  
Searching with share rights:  
 - 'license' - notuser -  (result: 0, expected: []) ok  
 - 'license' - User number_2 -  (result: 1, expected: ["license"]) ok  
 - 'license' - User3 -  (result: 1, expected: ["license"]) ok  
 - 'license' - User@4 -  (result: 1, expected: ["license"]) ok  
Removing test. ok  
Unlocking process ok  
 /var/www/html # `

**Comment by szaimen** [comment, marked as answer]
> Hi, see https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-run-occ-commands

Based on that the correct command is `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ fulltextsearch:index`

**Reply by caplam** [reply]
Thank you i have learned something.
I didn't know you have to run the command from the host. I was running "sudo -u www-data php occ fulltextsearch:index"
from inside the "nextcloud-aio-nextcloud" container.
I suppose the context is not the same.
Can i expect that after the first run it well be added to the cron or do i have to do something ?
For now it's still running but it already produces its effects even for the one file i threw to ocr.

edit: confirmed. A new doc is automatically indexed.
Now i have to find a way to ocr existing files without tagging them one by one.
